src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2386,8 +2389,23 @@ because xbuild doesn't support framework reference assemblies.
   </GetAddOnPlatformLibraries>
 </Target>
 
+<Target Name="_SetupAssemblyPreload"
+    Condition=" '$(AndroidEnablePreloadAssemblies)' == 'True' "
+    Inputs="$(IntermediateOutputPath)javastubs.cache"
```

Comment:
I don't immediately understand why this would use `$(IntermediateOutputPath)javastubs.cache` as an input...though that *is* what `_SetupEmbeddedDSOs` does...

Reply:
I don't think an appropriate `Inputs`/`Outputs` exists for this target, since there isn't a file that would influence what this does. I think we could just use `<WriteLinesToFile WriteOnlyWhenDifferent="True"/>` and not use any `Inputs`/`Outputs`. [msbuild docs](https://docs.microsoft.com/en-us/visualstudio/msbuild/writelinestofile-task?view=vs-2017)

`_GenerateJavaStubs` writes `javastubs.cache` looking at `android:extractNativeLibs="false"`, which determines the value of `$(_EmbeddedDSOsEnabled)`. So that makes sense for `_SetupEmbeddedDSOs` to use `javastubs.cache` as an input.